src/hotspot/share/cds/filemap.cpp

```diff
@@ -462,6 +462,11 @@ bool SharedClassPathEntry::validate(bool is_class_path) const {
     } else {
       FileMapInfo::fail_continue("A jar file is not the one used while building"
                                  " the shared archive file: %s", name);
+      if (_timestamp != st.st_mtime) {
+        log_warning(cds)("%s timestamp has changed.", name);
+      } else {
+        log_warning(cds)("%s size has changed.", name);
```

Comment:
Shouldn't these be incorporated into the message the fail_continue will report under log_info? By themselves the log_warning output will look odd, with no context.

Reply:
Thanks for the review. I've added the message from the fail_continue to the warning in case CDS logging is not enabled. Tests are also updated to look for the warning message.

Reply:
Background:

When -Xshare:auto is used, `FileMapInfo::fail_continue()` prints the diagnostic message to the Info channel instead of Warning. This was necessary in the past because CDS would failed to load for many reasons (e.g., failure to mmap due to ASLR) that are not the fault of the user. On Windows the failure rate could be as much as 50%. If we printed the message in the Warning channel, the user would be overwhelmed and would ultimately ignore the warnings, rendering them useless.

However, currently CDS is much more reliable. We should consider changing the failure logs to use the Warning channel instead.

I have filed [JDK-8292269](https://bugs.openjdk.org/browse/JDK-8292269).